test/test.py

```diff
@@ -8,29 +8,28 @@
 import io as _io
 import os as _os
 import re as _re
-import tempfile
 import multiprocessing
 import subprocess
 import unittest
 import mailbox
 import http.server
 import time
 import sys
+from pathlib import Path
+from typing import List
 
-# Directory containing test feed data/configs
-test_dir = _os.path.dirname(_os.path.abspath(__file__))
+from util.execcontext import r2e_path, ExecContext
+from util.tempmaildir import TemporaryMaildir
 
-# By default, we run the r2e in the dir above this script, not the
-# system-wide installed version, which you probably don't mean to
-# test. You can also pass in an alternate location in the R2E_PATH
-# environment variable.
-r2e_path = _os.getenv("R2E_PATH", _os.path.join(test_dir, "..", "r2e"))
+# Directory containing test feed data/configs
+test_dir = str(Path(__file__).absolute().parent.joinpath("data"))
 
 # Ensure we import the local (not system-wide) rss2email module
 sys.path.insert(0, _os.path.dirname(r2e_path))
```

Comment:
With this change, the import path, that was before
`_os.getenv("R2E_PATH", _os.path.join(_os.path.dirname(_os.path.abspath(__file__)), "..", "r2e"))`
, is now
`os.getenv("R2E_PATH", str(Path(__file__).absolute().parent.parent.parent.joinpath("r2e")))`
, which is one `.parent` more than before.

I haven't investigated much further, but would bet this is the source of the issue that happens inside nix.

If I understand correctly, the meaning of the `R2E_PATH` has changed compared to what it was before. Before it was pointing to the executable. Now it is pointing to… the library? I'm far from being sure about it, as I haven't reviewed in enough depth.

If it's just changing the meaning of the R2E_PATH to point to the library instead, then it'd be possible to simply adjust the `R2E_PATH` variable in `nix/release.nix` to make things work again.

However… Seeing as it's no longer an executable, but the python library, referring to its location by its relative path to the current file should just work, and the whole `R2E_PATH` (which was a hack designed to be able to call the binary from the tests) could be removed altogether. Which should hopefully solve the issue :) (you can check that by just removing the `R2E_PATH` assignment in `nix/release.nix`, and the `os.getenv` will then follow the same path on nix as on non-nix)

cc @Profpatsch just in case, as I think you probably introduced R2E_PATH?

Reply:
Actually, I was the one who introduced R2E_PATH. R2E_PATH should be unnecessary, since the test script should be able to find the `r2e` binary from the `test/` directory using `../r2e`, but this wasn't working for some reason.

Reply:
The value of `r2e_path` is the same. There is one `.parent` more, because the new `__file__` has one nested directory more than the previous one: `test/util/execcontext.py` vs `test/test.py`.

Reply:
@kaashif Oh, sorry for being mistaken about that!

@AndreyMZ Hmm… so if it doesn't point to the module but to the executable indeed, then the source of the issue is the changes like
```diff
-command = [r2e_path, "-VVVVV",
+command = [sys.executable, r2e_path, "-VVVVV",
```
Where you're calling `python {r2e executable}`, which isn't correct if r2e is being wrapped for any reason. And seeing as `R2E_PATH` was designed to take the wrapped r2e, this can't work.

So… just removing the `R2E_PATH` environment variable (both from `nix/release.nix` and the `os.getenv` in `test/util/execcontext.py`) should be enough to solve the issue, seeing as here we'd be specifically wanting a non-wrapped version of r2e? Either that, or revert the changes to the commands run, so that they don't call `python {r2e exec}` but directly `{r2e exec}`.

What do you think about these solutions?

Reply:
> revert the changes to the commands run, so that they don't call `python {r2e exec}` but directly `{r2e exec}`.

This would prevent running the tests on Windows.

>  just removing the R2E_PATH environment variable

Given that nobody needs `R2E_PATH` environment variable, I think this is the best solution.

I was hoping that I wouldn't have to deal with the configuration of Travis CI and Nix by myself... OK, I tried, but I have not succeed in running this CI locally. So, I have no idea how to run/debug it after changing `nix/release.nix` before commit. (I prefer not to commit the code which I never run.) Could someone help me with this, please?

Reply:
Of course! Once nix is installed ([instructions here](https://nixos.org/download.html)), you can just run `nix-build nix/release.nix -A pythonVersions` :)

Reply:
Done. I have removed the `R2E_PATH` environment variable, see https://github.com/rss2email/rss2email/pull/112/commits/c43715d2c86c8105d2b410ffb91a6cf3a3a7b400.